execution/process.go

```diff
@@ -197,59 +198,19 @@ func (p *ProcessByKey) MarkEndOfStream(ctx context.Context, tx storage.StateTran
 	return nil
 }
 
-func (p *ProcessByKey) Close() error {
-	return nil // TODO: Close this, remove state
-}
-
-type OutputQueue struct {
-	tx storage.StateTransaction
-}
-
-func NewOutputQueue(tx storage.StateTransaction) *OutputQueue {
-	return &OutputQueue{
-		tx: tx,
-	}
-}
-
-var queueElementsPrefix = []byte("$queue_elements$")
-
-func (q *OutputQueue) Push(ctx context.Context, element proto.Message) error {
-	queueElements := storage.NewDeque(q.tx.WithPrefix(queueElementsPrefix))
-
-	err := queueElements.PushBack(element)
+func (p *ProcessByKey) MarkError(ctx context.Context, tx storage.StateTransaction, err error) error {
+	outputQueue := NewOutputQueue(tx.WithPrefix(outputQueuePrefix))
+	err = outputQueue.Push(ctx, &QueueElement{Type: &QueueElement_Error{Error: err.Error()}})
```

Comment:
Can you indent QueueElement here like You did in kafka for example? It looks better imo